src/Illuminate/Testing/Concerns/TestDatabases.php

```diff
@@ -146,22 +146,24 @@ protected function whenNotUsingInMemoryDatabase($callback)
      */
     protected function switchToDatabase($database)
     {
-        DB::purge();
-
-        $default = config('database.default');
-
-        $url = config("database.connections.{$default}.url");
-
-        if ($url) {
-            config()->set(
-                "database.connections.{$default}.url",
-                preg_replace('/^(.*)(\/[\w-]*)(\??.*)$/', "$1/{$database}$3", $url),
-            );
-        } else {
-            config()->set(
-                "database.connections.{$default}.database",
-                $database,
-            );
+        $connections = config('database.testing.connections', [config('database.default')]);
+
+        foreach ($connections as $connection) {
```

Comment:
@ibrasho it doesn't appear that this sets database names for different connections. For example if my main database is `laravel` and I have another connection with the database name of `alternate`, does this overwrite the database name to be `laravel_test_1` for both connections? Shouldn't the alternate connection have a database name of `alternate_test_1`.

Reply:
I'm using Postgres schemas for the landlord and multiple tenants. I'm sure in other cases the requirements would be different.

What you are suggesting could also work, I'll try to see if I can make the logic simple and versatile at the same time.